service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/cmo/UploadCMOController.java

```diff
@@ -0,0 +1,147 @@
+
+        if (!(dynamicList instanceof DynamicList)) {
+            // reconstruct dynamic list
+            data.put("pastHearingList", cmoService.buildDynamicList(hearings, selectedHearing));
+        }
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(data)
+            .build();
+    }
+
+    @PostMapping("/about-to-submit")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody CallbackRequest request) {
+        CaseDetails caseDetails = request.getCaseDetails();
+        Map<String, Object> data = caseDetails.getData();
+        CaseData caseData = mapper.convertValue(data, CaseData.class);
+        List<Element<CaseManagementOrder>> draftCMOs = caseData.getDraftUploadedCMOs();
+        List<Element<HearingBooking>> hearingsWithoutCMO = cmoService.getHearingsWithoutCMO(
+            caseData.getPastHearings(), draftCMOs
+        );
+
+        if (!hearingsWithoutCMO.isEmpty()) {
+            List<Element<HearingBooking>> hearings = caseData.getHearingDetails();
+            UUID selectedHearingId = cmoService.getSelectedHearingId(caseData.getPastHearingList(),
+                hearingsWithoutCMO);
+            HearingBooking hearing = cmoService.getSelectedHearing(selectedHearingId, hearingsWithoutCMO);
+
+            Element<CaseManagementOrder> element = element(from(caseData.getUploadedCaseManagementOrder(),
+                hearing, time.now().toLocalDate()));
+
+            UUID uuid = cmoService.mapToHearing(selectedHearingId, hearings, element);
+
+            if (uuid != null) {
+                // overwrite old draft CMO
+                int index = -1;
+                for (int i = 0; i < draftCMOs.size(); i++) {
+                    if (draftCMOs.get(i).getId().equals(uuid)) {
+                        index = i;
+                        break;
+                    }
+                }
+
+                draftCMOs.set(index, element);
+            } else {
+                draftCMOs.add(element);
+            }
+
+            // update case data
+            data.put("draftUploadedCMOs", draftCMOs);
+            data.put("hearingDetails", hearings);
+        }
+        // remove transient fields
+        removeTemporaryFields(caseDetails, TRANSIENT_FIELDS);
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(data)
+            .build();
+    }
+
+    @PostMapping("/submitted")
+    public void handelSubmitted(@RequestBody CallbackRequest request) {
+        CaseData caseDataBefore = mapper.convertValue(request.getCaseDetailsBefore().getData(), CaseData.class);
+        CaseData caseData = mapper.convertValue(request.getCaseDetails().getData(), CaseData.class);
+
+        List<Element<CaseManagementOrder>> draftCMOs = caseData.getDraftUploadedCMOs();
+        List<Element<CaseManagementOrder>> draftCMOsBefore = caseDataBefore.getDraftUploadedCMOs();
+        draftCMOs.removeAll(draftCMOsBefore);
+
+        if (draftCMOs.size() == 1) {
```

Comment:
This is a bit logic, could it be abstracted to the service? It could be probably something like if(cmoService.anyNewCmoUploaded()) or something like that?